aten/src/ATen/native/cuda/ForeachReduceOp.cu

```diff
@@ -61,7 +61,11 @@ struct LpNormFunctor {
 #pragma unroll
         for (int ii = 0; ii < kILP; ii++) {
           opmath_t next = static_cast<opmath_t>(r_x[ii]);
-          vals[ii] += NormType == 1 ? ::abs(next) : next * next;
+          if constexpr (norm_type == NormType::LInf) {
+            vals[ii] = ::max(vals[ii], ::abs(next));
```

Comment:
std::max doesn't necessarily propagate nans, does it? (iirc in case of nan it returns first arg

Reply:
You are correct!